completion/services.py

```diff
@@ -118,7 +118,7 @@ def vertical_is_complete(self, item):
             return None
 
         # this is temporary local logic and will be removed when the whole course tree is included in completion
-        child_locations = [child.scope_ids.usage_id for child in self.get_completable_children(item)]
+        child_locations = [child.scope_ids.usage_id for child in self.get_completable_children(item) if child]
```

Comment:
This is actual surprising to me that this is somehow returning None. I think I might disagree with this being the fix. I can understand it as a band-aid, but this implies that somewhere, we have a call to an `xblock.get_child_descriptors` or `xblock.get_children` returning a list with None in it (`[None]`) and that feels more incorrect to me. Especially because we are only checking children when a block is marked as an Aggregator, indicating that it _should_ have children. I'm curious if others have thoughts on this and if we should perhaps take a look into those functions and why they might be returning a list of None.

Reply:
Agreed, this is a band-aid, not a solution.

I suspect the problem is related to the `XBlockCompletionMode.get_mode` calls [1],
but I don't think it's an issue with the `Aggregator`; instead I think it's with the leaf nodes.

> Especially because we are only checking children when a block is marked as an Aggregator, indicating that it should have children

Specifically, it's not that a block has _no_ children, but that a block has _one_ _or_ _more_ children whose value is `None`.

The final `get_mode` check is looking for `COMPLETABLE`:
but... this is the default value.

Given that [2], this reveals that the system will consider `None` to have a mode of `COMPLETABLE`, which doesn't seem correct. 

```python
XBlockCompletionMode.get_mode(None) == XBlockCompletionMode.COMPLETABLE
```

I assume the `hasattr` check is there to safeguard against blocks lacking this attribute, but since it doesn't check if it even has a `non-None` block in the first place, it "politely" falls back to the default... which probably isn't what we want here.

I have two more related commits that I'll push next:
- one cleans up `get_completable_children` to make it more clear what's happening
- the other inserts a `Truthy` check before checking for a leaf node's mode.

_but_ again, this doesn't answer the question of "why does it think it has a node that comes back `None`"?!
My first guess would be permissions related (so it "knows" it has a child, but then the user has no access to the child), but that's just a gut feeling...

- [1] https://github.com/edx/XBlock/blob/master/xblock/completable.py#L6-L21
- [2] https://github.com/edx/XBlock/blob/master/xblock/completable.py#L21

Reply:
FWIW: `XBlockCompletionMode.get_mode` _could_ first check for `None` and default to `EXCLUDED` in that case (instead of `COMPLETABLE`), though this _could_ introduce regressions in dependent code.

Reply:
Ah okay, yeah, your hunch about permissions certainly feels like a possible thing. And I agree with your hesitancy to make changes elsewhere that could have more downstream effects.

I'm okay with these changes then. One small question about if we still need this `if child` check here given your other changes. Approving either way.

Reply:
I also suspect the `if` check is no longer needed; I removed it.

I also bumped the fix/patch version.

Commits are squashed, waiting on the build.

Anything else I need to do pre/post merge?

Reply:
It looks like adding in a CHANGELOG entry and yes, bumping the version. I also might say adding in a small comment in the code for adding in this check and why it's necessary, but that's a nit.

I'm still seeing the commits, but that might just be a delay or something.

Post merge, you need to create the tag so the new version is pushed to PyPI and then update edx-platform for pulling in this newest version so we are actually applying the fix. So you can monitor PyPI for when this newest version is put up there and then add the new version into your platform PR. I think that'll do it.